[CI] Repropducible Failure in org.elasticsearch.upgrades.TransformSurvivesUpgradeIT

This reproduces locally and was just hit in a PR: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+bwc/8039/consoleFull


```
./gradlew ':x-pack:qa:rolling-upgrade:v8.0.0#oldClusterTest' -Dtests.class="org.elasticsearch.upgrades.TransformSurvivesUpgradeIT" -Dtests.method="testTransformRollingUpgrade" -Dtests.seed=505FFBF5C5C36777 -Dtests.bwc=true -Dtests.locale=it-IT -Dtests.timezone=America/Santiago -Druntime.java=11
```

fails as:

```

org.elasticsearch.upgrades.TransformSurvivesUpgradeIT > testTransformRollingUpgrade FAILED
    java.lang.AssertionError: Transform index templates not found. The templates that exist are: [.deprecation-indexing-template, .ml-anomalies-, .ml-notifications-000002, .ml-state, .ml-stats, .monitoring-alerts-7, .monitoring-beats, .monitoring-es, .monitoring-kibana, .monitoring-logstash, .slm-history, .watch-history-14, ilm-history, logs, metrics, synthetics]
        at __randomizedtesting.SeedInfo.seed([505FFBF5C5C36777:FF22CE4C6E85B4B3]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.upgrades.TransformSurvivesUpgradeIT.lambda$waitForTemplates$3(TransformSurvivesUpgradeIT.java:98)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:965)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:938)
        at org.elasticsearch.upgrades.TransformSurvivesUpgradeIT.waitForTemplates(TransformSurvivesUpgradeIT.java:81)
REPRODUCE WITH: ./gradlew ':x-pack:qa:rolling-upgrade:v8.0.0#oldClusterTest' -Dtests.class="org.elasticsearch.upgrades.TransformSurvivesUpgradeIT" -Dtests.method="testTransformRollingUpgrade" -Dtests.seed=505FFBF5C5C36777 -Dtests.bwc=true -Dtests.locale=it-IT -Dtests.timezone=America/Santiago -Druntime.java=11

Suite: Test class org.elasticsearch.upgrades.TransformSurvivesUpgradeIT
  1> [2021-10-15T07:10:35,513][INFO ][o.e.u.TransformSurvivesUpgradeIT] [testTransformRollingUpgrade] before test
  1> [2021-10-15T07:10:35,549][INFO ][o.e.u.TransformSurvivesUpgradeIT] [testTransformRollingUpgrade] initializing REST clients against [http://[::1]:34733, http://127.0.0.1:45697, http://[::1]:38297, http://127.0.0.1:37417, http://[::1]:46103, http://127.0.0.1:40407]
  1> [2021-10-15T07:10:46,309][INFO ][o.e.u.TransformSurvivesUpgradeIT] [testTransformRollingUpgrade] after test
  2> REPRODUCE WITH: ./gradlew ':x-pack:qa:rolling-upgrade:v8.0.0#oldClusterTest' -Dtests.class="org.elasticsearch.upgrades.TransformSurvivesUpgradeIT" -Dtests.method="testTransformRollingUpgrade" -Dtests.seed=505FFBF5C5C36777 -Dtests.bwc=true -Dtests.locale=it-IT -Dtests.timezone=America/Santiago -Druntime.java=11
  2> java.lang.AssertionError: Transform index templates not found. The templates that exist are: [.deprecation-indexing-template, .ml-anomalies-, .ml-notifications-000002, .ml-state, .ml-stats, .monitoring-alerts-7, .monitoring-beats, .monitoring-es, .monitoring-kibana, .monitoring-logstash, .slm-history, .watch-history-14, ilm-history, logs, metrics, synthetics]
        at __randomizedtesting.SeedInfo.seed([505FFBF5C5C36777:FF22CE4C6E85B4B3]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.upgrades.TransformSurvivesUpgradeIT.lambda$waitForTemplates$3(TransformSurvivesUpgradeIT.java:98)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:965)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:938)
        at org.elasticsearch.upgrades.TransformSurvivesUpgradeIT.waitForTemplates(TransformSurvivesUpgradeIT.java:81)
  2> NOTE: leaving temporary files on disk at: /home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testrun/v8.0.0#oldClusterTest/temp/org.elasticsearch.upgrades.TransformSurvivesUpgradeIT_505FFBF5C5C36777-001
  2> NOTE: test params are: codec=Asserting(Lucene90): {}, docValues:{}, maxPointsInLeafNode=874, maxMBSortInHeap=6.741854380936889, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=it-IT, timezone=America/Santiago
  2> NOTE: Linux 5.13.0-19-generic amd64/Oracle Corporation 11.0.2 (64-bit)/cpus=48,threads=1,free=481896528,total=536870912
  2> NOTE: All tests run in this JVM: [TokenBackwardsCompatibilityIT, SearchableSnapshotsRollingUpgradeIT, RollupDateHistoUpgradeIT, IndexingIT, TransformSurvivesUpgradeIT]

Tests with failures:
 - org.elasticsearch.upgrades.TransformSurvivesUpgradeIT.testTransformRollingUpgrade

1 test completed, 1 failed

=== Log output of node `node{:x-pack:qa:rolling-upgrade:v8.0.0-0}` ===

»    ↓ errors and warnings from /home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-0/logs/v8.0.0.log ↓
» [2021-10-15T12:10:27,425][WARN ][o.e.b.BootstrapChecks    ] [v8.0.0-0] HTTPS is required in order to use the token service; please enable HTTPS using the [xpack.security.http.ssl.enabled] setting or disable the token service using the [xpack.security.authc.token.enabled] setting
» [2021-10-15T12:10:32,433][WARN ][o.e.d.FileBasedSeedHostsProvider] [v8.0.0-0] expected, but did not find, a dynamic hosts list at [/home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-0/config/unicast_hosts.txt]
»   ↑ repeated 6 times ↑
»   ↓ last 40 non error or warning messages from /home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-0/logs/v8.0.0.log ↓
» [2021-10-15T12:10:22,274][INFO ][o.e.p.PluginsService     ] [v8.0.0-0] loaded module [x-pack-sql]
» [2021-10-15T12:10:22,274][INFO ][o.e.p.PluginsService     ] [v8.0.0-0] loaded module [x-pack-stack]
» [2021-10-15T12:10:22,274][INFO ][o.e.p.PluginsService     ] [v8.0.0-0] loaded module [x-pack-text-structure]
» [2021-10-15T12:10:22,275][INFO ][o.e.p.PluginsService     ] [v8.0.0-0] loaded module [x-pack-voting-only-node]
» [2021-10-15T12:10:22,275][INFO ][o.e.p.PluginsService     ] [v8.0.0-0] loaded module [x-pack-watcher]
» [2021-10-15T12:10:22,275][INFO ][o.e.p.PluginsService     ] [v8.0.0-0] no plugins loaded
» [2021-10-15T12:10:22,305][INFO ][o.e.e.NodeEnvironment    ] [v8.0.0-0] using [1] data paths, mounts [[/ (/dev/nvme0n1p2)]], net usable_space [439.1gb], net total_space [915.3gb], types [ext4]
» [2021-10-15T12:10:22,306][INFO ][o.e.e.NodeEnvironment    ] [v8.0.0-0] heap size [494.9mb], compressed ordinary object pointers [true]
» [2021-10-15T12:10:22,321][INFO ][o.e.n.Node               ] [v8.0.0-0] node name [v8.0.0-0], node ID [lVT9Fr_RTs27aHCbih2O5g], cluster name [v8.0.0], roles [ingest, data_frozen, ml, data_hot, transform, data_content, data_warm, master, remote_cluster_client, data, data_cold]
» [2021-10-15T12:10:25,219][INFO ][o.e.x.m.p.l.CppLogMessageHandler] [v8.0.0-0] [controller/402514] [Main.cc@123] controller (64 bit): Version 8.0.0-SNAPSHOT (Build f3909e0958c326) Copyright (c) 2021 Elasticsearch BV
» [2021-10-15T12:10:25,259][INFO ][o.e.x.s.Security         ] [v8.0.0-0] Security is enabled
» [2021-10-15T12:10:25,452][INFO ][o.e.x.s.a.Realms         ] [v8.0.0-0] license mode is [trial], currently licensed security realms are [reserved/reserved,file/file1,native/native1]
» [2021-10-15T12:10:25,458][INFO ][o.e.x.s.a.s.FileRolesStore] [v8.0.0-0] parsed [0] roles from file [/home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-0/config/roles.yml]
» [2021-10-15T12:10:25,955][INFO ][o.e.i.g.LocalDatabases   ] [v8.0.0-0] initialized config databases [[]] and watching [/home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-0/config/ingest-geoip] for changes
» [2021-10-15T12:10:25,957][INFO ][o.e.i.g.DatabaseRegistry ] [v8.0.0-0] initialized database registry, using geoip-databases directory [/home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-0/tmp/geoip-databases/lVT9Fr_RTs27aHCbih2O5g]
» [2021-10-15T12:10:26,451][INFO ][o.e.t.n.NettyAllocator   ] [v8.0.0-0] creating NettyAllocator with the following configs: [name=unpooled, suggested_max_allocation_size=1mb, factors={es.unsafe.use_unpooled_allocator=null, g1gc_enabled=false, g1gc_region_size=0b, heap_size=494.9mb}]
» [2021-10-15T12:10:26,497][INFO ][o.e.d.DiscoveryModule    ] [v8.0.0-0] using discovery type [zen] and seed hosts providers [settings, file]
» [2021-10-15T12:10:27,242][INFO ][o.e.n.Node               ] [v8.0.0-0] initialized
» [2021-10-15T12:10:27,242][INFO ][o.e.n.Node               ] [v8.0.0-0] starting ...
» [2021-10-15T12:10:27,251][INFO ][o.e.x.s.c.f.PersistentCache] [v8.0.0-0] persistent cache index loaded
» [2021-10-15T12:10:27,339][INFO ][o.e.t.TransportService   ] [v8.0.0-0] publish_address {127.0.0.1:45991}, bound_addresses {[::1]:38405}, {127.0.0.1:45991}
» [2021-10-15T12:10:27,445][INFO ][o.e.h.AbstractHttpServerTransport] [v8.0.0-0] publish_address {127.0.0.1:45697}, bound_addresses {[::1]:34733}, {127.0.0.1:45697}
» [2021-10-15T12:10:27,447][INFO ][o.e.n.Node               ] [v8.0.0-0] started
» [2021-10-15T12:10:33,652][INFO ][o.e.c.c.Coordinator      ] [v8.0.0-0] setting initial configuration to VotingConfiguration{lVT9Fr_RTs27aHCbih2O5g,{bootstrap-placeholder}-v8.0.0-2,xUnXAT5hQHWxNEpoHDJmnw}
» [2021-10-15T12:10:33,843][INFO ][o.e.c.c.CoordinationState] [v8.0.0-0] cluster UUID set to [TeOx3D61QDK7-gyyM9ROMA]
» [2021-10-15T12:10:33,862][INFO ][o.e.c.s.ClusterApplierService] [v8.0.0-0] master node changed {previous [], current [{v8.0.0-1}{xUnXAT5hQHWxNEpoHDJmnw}{WQjZSU7QSd6UGfy-ey8I-g}{127.0.0.1}{127.0.0.1:36803}{cdfhilmrstw}]}, added {{v8.0.0-1}{xUnXAT5hQHWxNEpoHDJmnw}{WQjZSU7QSd6UGfy-ey8I-g}{127.0.0.1}{127.0.0.1:36803}{cdfhilmrstw}}, term: 1, version: 1, reason: ApplyCommitRequest{term=1, version=1, sourceNode={v8.0.0-1}{xUnXAT5hQHWxNEpoHDJmnw}{WQjZSU7QSd6UGfy-ey8I-g}{127.0.0.1}{127.0.0.1:36803}{cdfhilmrstw}{testattr=test, ml.machine_memory=67273019392, ml.max_open_jobs=512, xpack.installed=true, ml.max_jvm_size=518979584}}
» [2021-10-15T12:10:33,941][INFO ][o.e.c.s.ClusterApplierService] [v8.0.0-0] added {{v8.0.0-2}{khnHASfdRMiDWVzl2DwkgQ}{gsihgsmPRSuVL1nAD-ZmgQ}{127.0.0.1}{127.0.0.1:39115}{cdfhilmrstw}}, term: 1, version: 2, reason: ApplyCommitRequest{term=1, version=2, sourceNode={v8.0.0-1}{xUnXAT5hQHWxNEpoHDJmnw}{WQjZSU7QSd6UGfy-ey8I-g}{127.0.0.1}{127.0.0.1:36803}{cdfhilmrstw}{testattr=test, ml.machine_memory=67273019392, ml.max_open_jobs=512, xpack.installed=true, ml.max_jvm_size=518979584}}
» [2021-10-15T12:10:34,882][INFO ][o.e.x.s.a.TokenService   ] [v8.0.0-0] refresh keys
» [2021-10-15T12:10:34,982][INFO ][o.e.x.s.a.TokenService   ] [v8.0.0-0] refreshed keys
» [2021-10-15T12:10:35,718][INFO ][o.e.l.LicenseService     ] [v8.0.0-0] license [d8f539a0-37db-4051-941e-98276d8c58fc] mode [trial] - valid
» [2021-10-15T12:10:35,718][INFO ][o.e.x.s.a.Realms         ] [v8.0.0-0] license mode is [trial], currently licensed security realms are [reserved/reserved,file/file1,native/native1]
» [2021-10-15T10:10:46.554799472Z] [BUILD] Stopping node

=== Log output of node `node{:x-pack:qa:rolling-upgrade:v8.0.0-1}` ===

»    ↓ errors and warnings from /home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-1/logs/v8.0.0.log ↓
» [2021-10-15T12:10:30,220][WARN ][o.e.b.BootstrapChecks    ] [v8.0.0-1] HTTPS is required in order to use the token service; please enable HTTPS using the [xpack.security.http.ssl.enabled] setting or disable the token service using the [xpack.security.authc.token.enabled] setting
» [2021-10-15T12:10:32,227][WARN ][o.e.d.FileBasedSeedHostsProvider] [v8.0.0-1] expected, but did not find, a dynamic hosts list at [/home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-1/config/unicast_hosts.txt]
»   ↑ repeated 3 times ↑
»   ↓ last 40 non error or warning messages from /home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-1/logs/v8.0.0.log ↓
» [2021-10-15T12:10:34,502][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding index template [.ml-state] for index patterns [.ml-state*]
» [2021-10-15T12:10:34,539][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding index template [.ml-notifications-000002] for index patterns [.ml-notifications-000002]
» [2021-10-15T12:10:34,591][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding index template [.ml-anomalies-] for index patterns [.ml-anomalies-*]
» [2021-10-15T12:10:34,628][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding index template [.ml-stats] for index patterns [.ml-stats-*]
» [2021-10-15T12:10:34,661][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding component template [logs-settings]
» [2021-10-15T12:10:34,698][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding component template [synthetics-mappings]
» [2021-10-15T12:10:34,730][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding component template [synthetics-settings]
» [2021-10-15T12:10:34,762][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding component template [logs-mappings]
» [2021-10-15T12:10:34,793][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding component template [metrics-mappings]
» [2021-10-15T12:10:34,824][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding component template [data-streams-mappings]
» [2021-10-15T12:10:34,851][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding component template [metrics-settings]
» [2021-10-15T12:10:35,001][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding index template [.watch-history-14] for index patterns [.watcher-history-14*]
» [2021-10-15T12:10:35,039][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding index template [ilm-history] for index patterns [ilm-history-5*]
» [2021-10-15T12:10:35,070][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding index template [.slm-history] for index patterns [.slm-history-5*]
» [2021-10-15T12:10:35,100][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding component template [.deprecation-indexing-mappings]
» [2021-10-15T12:10:35,129][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding component template [.deprecation-indexing-settings]
» [2021-10-15T12:10:35,162][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding index template [synthetics] for index patterns [synthetics-*-*]
» [2021-10-15T12:10:35,196][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding index template [logs] for index patterns [logs-*-*]
» [2021-10-15T12:10:35,237][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding index template [metrics] for index patterns [metrics-*-*]
» [2021-10-15T12:10:35,268][INFO ][o.e.c.m.MetadataIndexTemplateService] [v8.0.0-1] adding index template [.deprecation-indexing-template] for index patterns [.logs-deprecation.*]
» [2021-10-15T12:10:35,296][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [ml-size-based-ilm-policy]
» [2021-10-15T12:10:35,336][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [metrics]
» [2021-10-15T12:10:35,363][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [logs]
» [2021-10-15T12:10:35,387][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [synthetics]
» [2021-10-15T12:10:35,413][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [7-days-default]
» [2021-10-15T12:10:35,445][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [30-days-default]
» [2021-10-15T12:10:35,469][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [180-days-default]
» [2021-10-15T12:10:35,493][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [90-days-default]
» [2021-10-15T12:10:35,518][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [365-days-default]
» [2021-10-15T12:10:35,542][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [watch-history-ilm-policy]
» [2021-10-15T12:10:35,567][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [ilm-history-ilm-policy]
» [2021-10-15T12:10:35,589][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [slm-history-ilm-policy]
» [2021-10-15T12:10:35,613][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [.deprecation-indexing-ilm-policy]
» [2021-10-15T12:10:35,636][INFO ][o.e.x.i.a.TransportPutLifecycleAction] [v8.0.0-1] adding index lifecycle policy [.fleet-actions-results-ilm-policy]
» [2021-10-15T12:10:35,746][INFO ][o.e.l.LicenseService     ] [v8.0.0-1] license [d8f539a0-37db-4051-941e-98276d8c58fc] mode [trial] - valid
» [2021-10-15T12:10:35,746][INFO ][o.e.x.s.a.Realms         ] [v8.0.0-1] license mode is [trial], currently licensed security realms are [reserved/reserved,file/file1,native/native1]
» [2021-10-15T12:10:46,603][INFO ][o.e.t.ClusterConnectionManager] [v8.0.0-1] transport connection to [{v8.0.0-0}{lVT9Fr_RTs27aHCbih2O5g}{HfE1v9R5Q9OeF7EjI__KxQ}{127.0.0.1}{127.0.0.1:45991}{cdfhilmrstw}] closed by remote
» [2021-10-15T12:10:46,605][INFO ][o.e.c.s.MasterService    ] [v8.0.0-1] node-left[{v8.0.0-0}{lVT9Fr_RTs27aHCbih2O5g}{HfE1v9R5Q9OeF7EjI__KxQ}{127.0.0.1}{127.0.0.1:45991}{cdfhilmrstw} reason: disconnected], term: 1, version: 46, delta: removed {{v8.0.0-0}{lVT9Fr_RTs27aHCbih2O5g}{HfE1v9R5Q9OeF7EjI__KxQ}{127.0.0.1}{127.0.0.1:45991}{cdfhilmrstw}}
» [2021-10-15T10:10:46.607174890Z] [BUILD] Stopping node

=== Log output of node `node{:x-pack:qa:rolling-upgrade:v8.0.0-2}` ===

»    ↓ errors and warnings from /home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-2/logs/v8.0.0.log ↓
» [2021-10-15T12:10:32,987][WARN ][o.e.b.BootstrapChecks    ] [v8.0.0-2] HTTPS is required in order to use the token service; please enable HTTPS using the [xpack.security.http.ssl.enabled] setting or disable the token service using the [xpack.security.authc.token.enabled] setting
» [2021-10-15T12:10:32,994][WARN ][o.e.d.FileBasedSeedHostsProvider] [v8.0.0-2] expected, but did not find, a dynamic hosts list at [/home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-2/config/unicast_hosts.txt]
»   ↓ last 40 non error or warning messages from /home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-2/logs/v8.0.0.log ↓
» [2021-10-15T12:10:28,217][INFO ][o.e.p.PluginsService     ] [v8.0.0-2] loaded module [x-pack-security]
» [2021-10-15T12:10:28,217][INFO ][o.e.p.PluginsService     ] [v8.0.0-2] loaded module [x-pack-shutdown]
» [2021-10-15T12:10:28,217][INFO ][o.e.p.PluginsService     ] [v8.0.0-2] loaded module [x-pack-sql]
» [2021-10-15T12:10:28,217][INFO ][o.e.p.PluginsService     ] [v8.0.0-2] loaded module [x-pack-stack]
» [2021-10-15T12:10:28,217][INFO ][o.e.p.PluginsService     ] [v8.0.0-2] loaded module [x-pack-text-structure]
» [2021-10-15T12:10:28,217][INFO ][o.e.p.PluginsService     ] [v8.0.0-2] loaded module [x-pack-voting-only-node]
» [2021-10-15T12:10:28,218][INFO ][o.e.p.PluginsService     ] [v8.0.0-2] loaded module [x-pack-watcher]
» [2021-10-15T12:10:28,218][INFO ][o.e.p.PluginsService     ] [v8.0.0-2] no plugins loaded
» [2021-10-15T12:10:28,245][INFO ][o.e.e.NodeEnvironment    ] [v8.0.0-2] using [1] data paths, mounts [[/ (/dev/nvme0n1p2)]], net usable_space [439.1gb], net total_space [915.3gb], types [ext4]
» [2021-10-15T12:10:28,245][INFO ][o.e.e.NodeEnvironment    ] [v8.0.0-2] heap size [494.9mb], compressed ordinary object pointers [true]
» [2021-10-15T12:10:28,257][INFO ][o.e.n.Node               ] [v8.0.0-2] node name [v8.0.0-2], node ID [khnHASfdRMiDWVzl2DwkgQ], cluster name [v8.0.0], roles [ingest, data_frozen, ml, data_hot, transform, data_content, data_warm, master, remote_cluster_client, data, data_cold]
» [2021-10-15T12:10:30,955][INFO ][o.e.x.m.p.l.CppLogMessageHandler] [v8.0.0-2] [controller/403360] [Main.cc@123] controller (64 bit): Version 8.0.0-SNAPSHOT (Build f3909e0958c326) Copyright (c) 2021 Elasticsearch BV
» [2021-10-15T12:10:30,996][INFO ][o.e.x.s.Security         ] [v8.0.0-2] Security is enabled
» [2021-10-15T12:10:31,185][INFO ][o.e.x.s.a.Realms         ] [v8.0.0-2] license mode is [trial], currently licensed security realms are [reserved/reserved,file/file1,native/native1]
» [2021-10-15T12:10:31,190][INFO ][o.e.x.s.a.s.FileRolesStore] [v8.0.0-2] parsed [0] roles from file [/home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-2/config/roles.yml]
» [2021-10-15T12:10:31,688][INFO ][o.e.i.g.LocalDatabases   ] [v8.0.0-2] initialized config databases [[]] and watching [/home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-2/config/ingest-geoip] for changes
» [2021-10-15T12:10:31,689][INFO ][o.e.i.g.DatabaseRegistry ] [v8.0.0-2] initialized database registry, using geoip-databases directory [/home/brownbear/src/elasticsearch/x-pack/qa/rolling-upgrade/build/testclusters/v8.0.0-2/tmp/geoip-databases/khnHASfdRMiDWVzl2DwkgQ]
» [2021-10-15T12:10:32,102][INFO ][o.e.t.n.NettyAllocator   ] [v8.0.0-2] creating NettyAllocator with the following configs: [name=unpooled, suggested_max_allocation_size=1mb, factors={es.unsafe.use_unpooled_allocator=null, g1gc_enabled=false, g1gc_region_size=0b, heap_size=494.9mb}]
» [2021-10-15T12:10:32,144][INFO ][o.e.d.DiscoveryModule    ] [v8.0.0-2] using discovery type [zen] and seed hosts providers [settings, file]
» [2021-10-15T12:10:32,810][INFO ][o.e.n.Node               ] [v8.0.0-2] initialized
» [2021-10-15T12:10:32,810][INFO ][o.e.n.Node               ] [v8.0.0-2] starting ...
» [2021-10-15T12:10:32,818][INFO ][o.e.x.s.c.f.PersistentCache] [v8.0.0-2] persistent cache index loaded
» [2021-10-15T12:10:32,902][INFO ][o.e.t.TransportService   ] [v8.0.0-2] publish_address {127.0.0.1:39115}, bound_addresses {[::1]:46613}, {127.0.0.1:39115}
» [2021-10-15T12:10:33,006][INFO ][o.e.h.AbstractHttpServerTransport] [v8.0.0-2] publish_address {127.0.0.1:40407}, bound_addresses {[::1]:46103}, {127.0.0.1:40407}
» [2021-10-15T12:10:33,007][INFO ][o.e.n.Node               ] [v8.0.0-2] started
» [2021-10-15T12:10:33,813][INFO ][o.e.c.c.Coordinator      ] [v8.0.0-2] setting initial configuration to VotingConfiguration{lVT9Fr_RTs27aHCbih2O5g,khnHASfdRMiDWVzl2DwkgQ,xUnXAT5hQHWxNEpoHDJmnw}
» [2021-10-15T12:10:33,963][INFO ][o.e.c.s.ClusterApplierService] [v8.0.0-2] master node changed {previous [], current [{v8.0.0-1}{xUnXAT5hQHWxNEpoHDJmnw}{WQjZSU7QSd6UGfy-ey8I-g}{127.0.0.1}{127.0.0.1:36803}{cdfhilmrstw}]}, added {{v8.0.0-0}{lVT9Fr_RTs27aHCbih2O5g}{HfE1v9R5Q9OeF7EjI__KxQ}{127.0.0.1}{127.0.0.1:45991}{cdfhilmrstw}, {v8.0.0-1}{xUnXAT5hQHWxNEpoHDJmnw}{WQjZSU7QSd6UGfy-ey8I-g}{127.0.0.1}{127.0.0.1:36803}{cdfhilmrstw}}, term: 1, version: 2, reason: ApplyCommitRequest{term=1, version=2, sourceNode={v8.0.0-1}{xUnXAT5hQHWxNEpoHDJmnw}{WQjZSU7QSd6UGfy-ey8I-g}{127.0.0.1}{127.0.0.1:36803}{cdfhilmrstw}{testattr=test, ml.machine_memory=67273019392, ml.max_open_jobs=512, xpack.installed=true, ml.max_jvm_size=518979584}}
» [2021-10-15T12:10:34,882][INFO ][o.e.x.s.a.TokenService   ] [v8.0.0-2] refresh keys
» [2021-10-15T12:10:34,983][INFO ][o.e.x.s.a.TokenService   ] [v8.0.0-2] refreshed keys
» [2021-10-15T12:10:35,726][INFO ][o.e.l.LicenseService     ] [v8.0.0-2] license [d8f539a0-37db-4051-941e-98276d8c58fc] mode [trial] - valid
» [2021-10-15T12:10:35,727][INFO ][o.e.x.s.a.Realms         ] [v8.0.0-2] license mode is [trial], currently licensed security realms are [reserved/reserved,file/file1,native/native1]
» [2021-10-15T12:10:46,602][INFO ][o.e.t.ClusterConnectionManager] [v8.0.0-2] transport connection to [{v8.0.0-0}{lVT9Fr_RTs27aHCbih2O5g}{HfE1v9R5Q9OeF7EjI__KxQ}{127.0.0.1}{127.0.0.1:45991}{cdfhilmrstw}] closed by remote
» [2021-10-15T12:10:46,612][INFO ][o.e.c.s.ClusterApplierService] [v8.0.0-2] removed {{v8.0.0-0}{lVT9Fr_RTs27aHCbih2O5g}{HfE1v9R5Q9OeF7EjI__KxQ}{127.0.0.1}{127.0.0.1:45991}{cdfhilmrstw}}, term: 1, version: 46, reason: ApplyCommitRequest{term=1, version=46, sourceNode={v8.0.0-1}{xUnXAT5hQHWxNEpoHDJmnw}{WQjZSU7QSd6UGfy-ey8I-g}{127.0.0.1}{127.0.0.1:36803}{cdfhilmrstw}{testattr=test, ml.machine_memory=67273019392, ml.max_open_jobs=512, xpack.installed=true, ml.max_jvm_size=518979584}}
» [2021-10-15T12:10:46,655][INFO ][o.e.t.ClusterConnectionManager] [v8.0.0-2] transport connection to [{v8.0.0-1}{xUnXAT5hQHWxNEpoHDJmnw}{WQjZSU7QSd6UGfy-ey8I-g}{127.0.0.1}{127.0.0.1:36803}{cdfhilmrstw}] closed by remote
» [2021-10-15T12:10:46,658][INFO ][o.e.c.c.Coordinator      ] [v8.0.0-2] master node [{v8.0.0-1}{xUnXAT5hQHWxNEpoHDJmnw}{WQjZSU7QSd6UGfy-ey8I-g}{127.0.0.1}{127.0.0.1:36803}{cdfhilmrstw}] disconnected, restarting discovery
» [2021-10-15T10:10:46.658937626Z] [BUILD] Stopping node
» [2021-10-15T12:10:46,660][INFO ][o.e.c.s.ClusterApplierService] [v8.0.0-2] master node changed {previous [{v8.0.0-1}{xUnXAT5hQHWxNEpoHDJmnw}{WQjZSU7QSd6UGfy-ey8I-g}{127.0.0.1}{127.0.0.1:36803}{cdfhilmrstw}], current []}, term: 1, version: 46, reason: becoming candidate: onLeaderFailure

```

--- a/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TransformSurvivesUpgradeIT.java
+++ b/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TransformSurvivesUpgradeIT.java
@@ -78,8 +78,8 @@ public class TransformSurvivesUpgradeIT extends AbstractUpgradeTestCase {
 
     @Before
     public void waitForTemplates() throws Exception {
-        // no transform before 7.2
-        if (UPGRADE_FROM_VERSION.before(Version.V_7_2_0)) {
+        // no transform before 7.2, after 7.16 this check isn't required anymore
+        if (UPGRADE_FROM_VERSION.before(Version.V_7_2_0) || UPGRADE_FROM_VERSION.onOrAfter(Version.V_7_16_0)) {
             return;
         }
 

